src/pages/users/UsersPage.tsx

```diff
@@ -159,13 +159,57 @@ const UsersPage: FunctionComponent = () => {
                     headerId: 'users.table.error.delete',
                 })
             )
-            .then(() => gridContext?.refresh?.());
+            .then(() => {
+                // close the dialog
+                setShow(false);
+                gridContext?.refresh?.();
+            });
     }, [gridContext, rowsSelection, snackError]);
     const deleteUsersDisabled = useMemo(
         () => rowsSelection.length <= 0,
         [rowsSelection.length]
     );
 
+    const handleDeletion = (value: boolean) => {
```

Comment:
you can directly call show(DeletionDialog) instead of handleDeletion. It would be more descriptive.

Reply:
I always prefer to use a separate method instead of calling the setXX directly. This way  it's easy to maintain  and we can add other verification as well if needed.

Reply:
yes, it makes sense too.